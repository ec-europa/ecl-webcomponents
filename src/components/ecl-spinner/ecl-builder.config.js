const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-spinner-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-spinner-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-spinner-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-spinner-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
