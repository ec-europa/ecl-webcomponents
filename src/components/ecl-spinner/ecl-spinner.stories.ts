const getArgs = () => {
  return {
    variant: 'primary',
    label: 'Loading...',
    centered: true,
    visible: true,
    overlay: false,
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['primary', 'negative'],
      description: 'Spinner variant',
    },
    label: {
      type: { name: 'string' },
      description: 'Spinner label',
    },
    centered: {
      type: { name: 'boolean' },
      description: 'Centered spinner',
    },
    visible: {
      type: { name: 'boolean' },
      description: 'Visible spinner',
    },
    overlay: {
      type: { name: 'boolean' },
      description: 'with overlay',
    },
  };
};

export default {
  title: 'Components/spinner',
};

const Template = args => {
  let story = `<ecl-spinner
    variant="${args.variant}"
    theme="${args.theme}"
    centered=${args.centered}
    visible=${args.visible}
    overlay=${args.overlay}
  >
    ${args.label}
  </ecl-spinner>`;

  if (args.variant === 'negative') {
    story = `<div style="background: #004494; position: relative; height: 150px;">${story}</div>`
  }

  return story;
}


export const Spinner = Template.bind({});
Spinner.storyName = 'default';
Spinner.args = getArgs();
Spinner.argTypes = getArgTypes();