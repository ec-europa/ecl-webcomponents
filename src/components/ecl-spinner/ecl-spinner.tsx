import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-spinner',
  styleUrls: {
    ec: './build/styles/ecl-spinner-ec.css',
    eu: './build/styles/ecl-spinner-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclSpinner {
  @Prop() styleClass: string = '';
  @Prop() variant: string = 'primary';
  @Prop() theme: string = 'ec';
  @Prop() centered: boolean = false;
  @Prop() visible: boolean = false;
  @Prop() overlay: boolean = false;
  @Prop() size: string = 'm';

  getClass(): string {
    const classes = [
      `ecl-spinner`,
      `ecl-spinner--${this.variant}`,
      `ecl-spinner--${this.size}`,
      this.styleClass
    ];

    if (this.centered) {
      classes.push('ecl-spinner--centered');
    }
    if (this.visible) {
      classes.push('ecl-spinner--visible');
    }

    return classes.join(' ');
  }

  getOverlayClass(): string {
    const overlayClasses = ['ecl-spinner__overlay'];

    if (this.visible) {
      overlayClasses.push('ecl-spinner__overlay--visible');
    }

    return overlayClasses.join(' ');
  }

  render() {
    return (
    <div>
    { this.overlay ? 
      <div class={this.getOverlayClass()}></div> : '' 
    }
      <div class={this.getClass()}>
        <svg
          class="ecl-spinner__loader"
          viewBox="25 25 50 50"
        >
          <circle
            class="ecl-spinner__circle"
            cx="50"
            cy="50"
            r="20"
            fill="none"
            stroke-width="4px"
            stroke-miterlimit="10"
            vector-effect="non-scaling-stroke"
          />
        </svg>
        <div class="ecl-spinner__text">
          <slot></slot>
        </div>
      </div>
    </div>
    )
  }
}
