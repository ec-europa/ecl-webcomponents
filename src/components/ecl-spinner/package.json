{
  "name": "@ecl/web-components-spinner",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.15.1",
  "description": "ECL Spinner",
  "main": "ecl-spinner.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/vanilla-component-spinner": "3.13.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
