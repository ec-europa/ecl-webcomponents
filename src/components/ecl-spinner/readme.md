# ecl-spinner



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default     |
| ------------ | ------------- | ----------- | --------- | ----------- |
| `centered`   | `centered`    |             | `boolean` | `false`     |
| `overlay`    | `overlay`     |             | `boolean` | `false`     |
| `size`       | `size`        |             | `string`  | `'medium'`  |
| `styleClass` | `style-class` |             | `string`  | `''`        |
| `theme`      | `theme`       |             | `string`  | `'ec'`      |
| `variant`    | `variant`     |             | `string`  | `'primary'` |
| `visible`    | `visible`     |             | `boolean` | `false`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
