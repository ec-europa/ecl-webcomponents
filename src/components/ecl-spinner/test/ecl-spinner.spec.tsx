import { newSpecPage } from '@stencil/core/testing';
import { EclSpinner } from '../ecl-spinner';

describe('ecl-spinner', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclSpinner],
      html: `<ecl-spinner
        variant="primary"
        theme="ec"
        centered
        visible
      >
        Loading...
      </ecl-spinner>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});