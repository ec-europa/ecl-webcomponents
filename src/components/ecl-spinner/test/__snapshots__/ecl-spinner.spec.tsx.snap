// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-spinner renders 1`] = `
<ecl-spinner centered="" size="l" variant="primary" visible="">
  <!---->
  <div>
    <div class="ecl-spinner ecl-spinner--centered ecl-spinner--l ecl-spinner--primary ecl-spinner--visible">
      <svg class="ecl-spinner__loader" viewBox="25 25 50 50">
        <circle class="ecl-spinner__circle" cx="50" cy="50" fill="none" r="20" stroke-miterlimit="10" stroke-width="4px" vector-effect="non-scaling-stroke"></circle>
      </svg>
      <div class="ecl-spinner__text">
        Loading...
      </div>
    </div>
  </div>
</ecl-spinner>
`;
